Test suite fails

Probably a symptom of newer Rust toolchains, but the test suite currently fails 8 tests and eventually aborts for me using current Rust stable.

Sure. Attached is a build log using Rust 1.54.0. Note that I don't *anywhere* see "nightly" mentioned in the repo. Baring a `rust-toolchain` file that says "nightly" (or the new toml format) I always package software by building with the latest stable release of the compiler.

<details>
<summary>Details</summary>
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling syn v1.0.50
   Compiling serde_derive v1.0.117
   Compiling version_check v0.9.2
   Compiling bitflags v1.2.1
   Compiling memchr v2.3.4
   Compiling serde v1.0.117
   Compiling unicode-width v0.1.8
   Compiling pulldown-cmark v0.8.0 (/build/pulldown-cmark-git/src/pulldown-cmark)
   Compiling unicase v2.6.0
   Compiling getopts v0.2.21
warning: Generated 8 tests in "./tests/suite/footnotes.rs"
warning: Generated 67 tests in "./tests/suite/regression.rs"
warning: Generated 12 tests in "./tests/suite/table.rs"
warning: Generated 649 tests in "./tests/suite/spec.rs"
warning: Generated 16 tests in "./tests/suite/smart_punct.rs"
warning: Generated 8 tests in "./tests/suite/gfm_table.rs"
warning: Generated 2 tests in "./tests/suite/gfm_strikethrough.rs"
warning: Generated 2 tests in "./tests/suite/gfm_tasklist.rs"
   Compiling quote v1.0.7
error[E0453]: allow(unsafe_code) incompatible with previous forbid
  --> src/lib.rs:56:37
   |
55 | #![forbid(unsafe_code)]
   |           ----------- `forbid` level set here
56 | #![cfg_attr(feature = "simd", allow(unsafe_code))]
   |                                     ^^^^^^^^^^^ overruled by previous forbid

error: usage of an `unsafe` block
   --> src/escape.rs:210:13
    |
210 | /             unsafe {
211 | |                 foreach_special_simd(bytes, 0, |i| {
212 | |                     let escape_ix = *bytes.get_unchecked(i) as usize;
213 | |                     let replacement =
...   |
219 | |                 w.write_str(&s.get_unchecked(mark..))
220 | |             }
    | |_____________^
    |
note: the lint level is defined here
   --> src/lib.rs:55:11
    |
55  | #![forbid(unsafe_code)]
    |           ^^^^^^^^^^^

error: declaration of an `unsafe` function
   --> src/escape.rs:243:5
    |
243 | /     unsafe fn compute_mask(bytes: &[u8], offset: usize) -> i32 {
244 | |         debug_assert!(bytes.len() >= offset + VECTOR_SIZE);
245 | |
246 | |         let table = create_lookup();
...   |
268 | |         _mm_movemask_epi8(matches)
269 | |     }
    | |_____^

error: declaration of an `unsafe` function
   --> src/escape.rs:276:5
    |
276 | /     unsafe fn foreach_special_simd<F>(
277 | |         bytes: &[u8],
278 | |         mut offset: usize,
279 | |         mut callback: F,
...   |
314 | |         Ok(())
315 | |     }
    | |_____^

error: declaration of an `unsafe` function
    --> src/firstpass.rs:1559:5
     |
1559 | /     unsafe fn compute_mask(lut: &[u8; 16], bytes: &[u8], ix: usize) -> i32 {
1560 | |         debug_assert!(bytes.len() >= ix + VECTOR_SIZE);
1561 | |
1562 | |         let bitmap = _mm_loadu_si128(lut.as_ptr() as *const __m128i);
...    |
1591 | |         _mm_movemask_epi8(result)
1592 | |     }
     | |_____^

error: usage of an `unsafe` block
    --> src/firstpass.rs:1608:13
     |
1608 |             unsafe { simd_iterate_special_bytes(&lut.simd, bytes, ix, callback) }
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: declaration of an `unsafe` function
    --> src/firstpass.rs:1618:5
     |
1618 | /     unsafe fn process_mask<F, T>(
1619 | |         mut mask: i32,
1620 | |         bytes: &[u8],
1621 | |         mut offset: usize,
...    |
1638 | |         Ok(offset)
1639 | |     }
     | |_____^

error: declaration of an `unsafe` function
    --> src/firstpass.rs:1644:5
     |
1644 | /     unsafe fn simd_iterate_special_bytes<F, T>(
1645 | |         lut: &[u8; 16],
1646 | |         bytes: &[u8],
1647 | |         mut ix: usize,
...    |
1673 | |         (bytes.len(), None)
1674 | |     }
     | |_____^

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0453`.
The following warnings were emitted during compilation:

warning: Generated 8 tests in "./tests/suite/footnotes.rs"
warning: Generated 67 tests in "./tests/suite/regression.rs"
warning: Generated 12 tests in "./tests/suite/table.rs"
warning: Generated 649 tests in "./tests/suite/spec.rs"
warning: Generated 16 tests in "./tests/suite/smart_punct.rs"
warning: Generated 8 tests in "./tests/suite/gfm_table.rs"
warning: Generated 2 tests in "./tests/suite/gfm_strikethrough.rs"
warning: Generated 2 tests in "./tests/suite/gfm_tasklist.rs"

error: could not compile `pulldown-cmark`

To learn more, run the command again with --verbose.
[1m[31m==> ERROR:[m[1m A failure occurred in build().[m
[1m    Aborting...[m

</details>

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -52,8 +52,7 @@
 #![cfg_attr(rustbuild, feature(staged_api, rustc_private))]
 #![cfg_attr(rustbuild, unstable(feature = "rustc_private", issue = "27812"))]
 // Forbid unsafe code unless the SIMD feature is enabled.
-#![forbid(unsafe_code)]
-#![cfg_attr(feature = "simd", allow(unsafe_code))]
+#![cfg_attr(not(feature = "simd"), forbid(unsafe_code))]
 
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};

